### PR TITLE
fix: fix metrics trackEvent compatibility with legacy events

### PR DESCRIPTION
**Pull request by NicolasMassart**
## **Description**

Add handling of legacy events that have properties inside the event

## **Related issues**

Realted to https://github.com/MetaMask/mobile-planning/issues/1129

## **Manual testing steps**

NA - see unit tests

### **Before**

NA

### **After**

NA

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by tommasini — DISMISSED** [review]
LGTM!

File: app/core/Analytics/MetaMetrics.ts [hunk 1]
```diff
@@ -601,7 +601,7 @@ class MetaMetrics implements IMetaMetrics {
     if (this.enabled) {
       this.#trackEvent(
         event.category,
-        { anonymous: false, ...properties },
+        { anonymous: false, ...event.properties, ...properties },
```
[the remark this answers is not in the document]
Comment (NicolasMassart) [reply app/core/Analytics/MetaMetrics.ts:604]: Oh, JS files... I alwasy forget obout them...

[the remark this answers is not in the document]
Comment (NicolasMassart) [reply app/core/Analytics/MetaMetrics.ts:604]:
```suggestion
        { anonymous: false, ...event?.properties, ...properties },
```

**Review by tommasini — DISMISSED** [review]
LGTM

**Review by tommasini — DISMISSED** [review]
LGTM!

**Review by tommasini — APPROVED** [review]

**Pull request merged**